docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device/remote-tensor-api-npu-plugin.rst

```diff
@@ -0,0 +1,136 @@
+
+Getting RemoteContext from the Plugin
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+To request the current default context of the plugin, use one of the following methods:
+
+.. tab-set::
+
+   .. tab-item:: Get context from Core
+      :sync: get-context-core
+
+      .. doxygensnippet:: docs/articles_en/assets/snippets/npu_remote_objects_creation.cpp
+          :language: cpp
+          :fragment: [default_context_from_core]
+
+   .. tab-item:: Get context from compiled model
+      :sync: get-context-compiled-model
+
+      .. doxygensnippet:: docs/articles_en/assets/snippets/npu_remote_objects_creation.cpp
+          :language: cpp
+          :fragment: [default_context_from_model]
+
+Memory Sharing Between Application and NPU Plugin
+###########################################################
+
+The classes that implement the ``ov::RemoteTensor`` interface are the wrappers for native API
+memory handles (which can be obtained from them at any time).
+
+To create a shared tensor from a native memory handle, use dedicated ``create_tensor``, ``create_l0_host_tensor``, or ``create_host_tensor``
+methods of the ``ov::RemoteContext`` sub-classes.
+``ov::intel_npu::level_zero::LevelZero`` has multiple overloads methods which allow to wrap pre-allocated native handles with the ``ov::RemoteTensor``
+object or request plugin to allocate specific device memory.
+For more details, see the code snippets below:
+
+
+.. tab-set::
+
+   .. tab-item:: Wrap native handles
+      :sync: wrap-native-handles
+
+      .. tab-set::
+
+         .. tab-item:: Direct3D12 resource
+            :sync: d3d12
+
+            .. doxygensnippet:: docs/articles_en/assets/snippets/npu_remote_objects_creation.cpp
+               :language: cpp
+               :fragment: [wrap_d3d12_resource]
+
+         .. tab-item:: DMA-BUF System Heap resource
+            :sync: dma-buf
+
+            .. doxygensnippet:: docs/articles_en/assets/snippets/npu_remote_objects_creation.cpp
+               :language: cpp
+               :fragment: [wrap_dmabuf_resource]
+
+   .. tab-item:: Allocate device memory
+      :sync: allocate-device-memory
+
+      .. tab-set::
+
+         .. tab-item:: Remote Tensor - Level Zero host memory
+            :sync: remote-level-zero-host-memory
+
+            .. doxygensnippet:: docs/articles_en/assets/snippets/npu_remote_objects_creation.cpp
+               :language: cpp
+               :fragment: [allocate_remote_level_zero_host]
+
+         .. tab-item:: Tensor - Level Zero host memory
+            :sync: level-zero-host-memory
+
+            .. doxygensnippet:: docs/articles_en/assets/snippets/npu_remote_objects_creation.cpp
+               :language: cpp
+               :fragment: [allocate_level_zero_host]
+
+
+Limitations
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+* Allocation of the Direct3D12 or DMA-BUF System Heap resources is user's responsibility.
```

Comment:
Creation of an NT handle or allocation of DMA-BUF...

Reply:
done, thanks!